### ISSUE TITLE
Does the dev behind nnn plan to allow users to create custom keybindings without the need to install packages?

### ISSUE THREAD

**Issue by Raytle** [jarun/nnn#1103]
I'm currently trying to answer the question: **is nnn right for me?** The following is neither a rant nor even a screed but instead a forthright, blunt, and even harsh explanation of my current thoughts and concerns about nnn.

I'm not an engineer; instead I'm a mere "power-user" who occasionally cobbles together some very simple Python and Bash scripts based on examples I find on StackOverflow. In other words, I suppose that I am probably must less technically sophisticated than the ordinary nnn user.

In [kitty](https://github.com/kovidgoyal/kitty) (a very popular file manager that I guess you have probably heard of) and [spacefm](https://github.com/IgnorantGuru/spacefm) (an essentially orphaned GUI file manager), it is trivial for me to add custom keybindings. I have added many custom keybindings to those applications over the years because, well, of course, those custom keybindings make sense **to me**. I consider those custom keybindings to be neither sane nor insane but rather merely matters of my own personal preference.

This issue [Sane keybinds and switches](https://github.com/jarun/nnn/issues/422) makes me reluctant to use nnn. And. Well. Frankly. This kind of glib, arrogance makes me cringe...

> [TL;DR: Can't install packages?](https://github.com/jarun/nnn/wiki/Usage#installation) Get the statically linked binary from the latest release.

Why not opine, "Because I failed to properly architect nnn to make it easy for users to create custom keybindings, I have painted myself into a corner on this issue. In other words, because of my bad architecture it would be too much work for me to make it easy for users to create custom keybindings. Therefore, I have latch on to the absurd notion that I have created, 'sane keybinds and switches' because, of course, any different custom keybindings a user might want are, by definition, insane."

In other words, because other devs have enabled users to easily add custom keybindings to their applications—without the need to build those applications but instead after merely installing those apps them from their distro's repo—then obviously the dev behind nnn could have done so as well. 

Sure. Yes. Of course. I can install packages. I have probably installed packages at least a thousand of by now. Over the years I've installed and reinstalled a dozen or so flavors of Linux, I would estimate around a hundred times by now. As a result I needed to reinstall the same applications many, many times.

But, what I have learned from using many different FOSS applications is this: devs who have **contempt for users**, devs who have failed to add simple, easy, and commonplace functionality that makes the user experience easier, are usually devs who so fixated on adding what they claim are "important" features—but which are typically merely unnecessary bells and whistles to simpletons like me—that those devs end up building an applications that they and a cohort of like-minded, well, hmmm, I'll say it, **nerds** find great but ordinary users (like me) find remarkably user-**unfriendly**.

Based on my experiences dealing with FOSS devs, I expect this issue to be quickly closed without a response because, of course, challenging a FOSS dev on his own turf results in soft ban.

**Comment by jarun** [comment]
> This issue Sane keybinds and switches makes me reluctant to use nnn. And. Well. Frankly. This kind of glib, arrogance makes me cringe...

I don't see why you have to use `nnn` if you find it challenging to use. There are tons of terminal file managers available and probably all of them support customisable keybinds without the need to compile the program. And some of them are really good. Please choose one among those.

**Comment by jarun** [comment]
> TL;DR: Can't install packages? Get the statically linked binary from the latest release.

That's intended for users with a permission problem who can't install packages. `nnn` doesn't support custom keybinds out of the box in any form.

**Comment by Raytle** [comment]
> I don't see why you have to use nnn if you find it challenging to use.

I don't intend to use nnn.


> There are tons of terminal file managers available and probably all of them support customisable keybinds without the need to compile the program.

That seems like a reasonable presumption.


> And some of them are really good.

I presume that is true.


> Please choose one among those.

I intend to do so.

For a guy—that would be you—who vigorously defended nnn against all comers (essentially all criticisms) on sites like HackerNews (Ycombinator) you folded up like a house of cards buffeted by a light breeze.

Whenever devs do functional architecture or user interface design, they invariably make foolish blunders. Failing to support customizable keybindings without the need to compile the nnn was a major blunder which you have decided to defend and ignore but not fix. I detest dealing with ignorant devs like you. I've dealt with throngs of engineers (devs being a type of engineer) who implicitly proclaim, "I am an engineer; I am always right!" Despite those criticisms, I wish you farewell.
